File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/calculo/honorarios/CredorVO.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.dominio.calculo.honorarios;

import br.jus.trt8.pjecalc.base.comum.Utils;
import br.jus.trt8.pjecalc.negocio.constantes.TipoDocumentoFiscalEnum;

public class CredorVO {
    private static final String NAO_INFORMADO = " n\u00e3o informado";
    private String nome;
    private TipoDocumentoFiscalEnum tipoDocumentoFiscalCredor;
    private String numeroDocumentoFiscalCredor;

    public String getNome() {
        return this.nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public TipoDocumentoFiscalEnum getTipoDocumentoFiscalCredor() {
        return this.tipoDocumentoFiscalCredor;
    }

    public void setTipoDocumentoFiscalCredor(TipoDocumentoFiscalEnum tipoDocumentoFiscalCredor) {
        this.tipoDocumentoFiscalCredor = tipoDocumentoFiscalCredor;
    }

    public String getNumeroDocumentoFiscalCredor() {
        return this.numeroDocumentoFiscalCredor;
    }

    public void setNumeroDocumentoFiscalCredor(String numeroDocumentoFiscalCredor) {
        this.numeroDocumentoFiscalCredor = numeroDocumentoFiscalCredor;
    }

    public String getRepresentacaoTextual() {
        StringBuilder sb = new StringBuilder();
        if (Utils.naoNulo(this.nome)) {
            sb.append(this.nome);
        } else {
            sb.append("Nome");
            sb.append(NAO_INFORMADO);
        }
        sb.append(" - ");
        if (Utils.naoNulo((Object)this.tipoDocumentoFiscalCredor)) {
            sb.append((Object)this.tipoDocumentoFiscalCredor);
        } else {
            sb.append("Tipo de Documento");
            sb.append(NAO_INFORMADO);
        }
        sb.append(": ");
        if (Utils.naoNulo(this.numeroDocumentoFiscalCredor)) {
            sb.append(this.numeroDocumentoFiscalCredor);
        } else {
            sb.append("N\u00famero de Documento");
            sb.append(NAO_INFORMADO);
        }
        return sb.toString();
    }

    public static CredorVO converteCredor(String selecao) {
        CredorVO credorVO = new CredorVO();
        if (Utils.naoNulo(selecao) && !selecao.equals("Preencher Dados...")) {
            String numeroDocumento;
            String tipoDocumento;
            String nome = selecao.split(" - ")[0];
            if (!nome.contains(NAO_INFORMADO)) {
                credorVO.setNome(nome);
            }
            if (!(tipoDocumento = selecao.split(" - ")[1].split(": ")[0]).contains(NAO_INFORMADO)) {
                credorVO.setTipoDocumentoFiscalCredor(TipoDocumentoFiscalEnum.valueOf(tipoDocumento));
            }
            if (!(numeroDocumento = selecao.split(" - ")[1].split(": ")[1]).contains(NAO_INFORMADO)) {
                credorVO.setNumeroDocumentoFiscalCredor(numeroDocumento);
            }
        }
        return credorVO;
    }
}

File: src/lib/pjecalc/core/dominio/calculo/honorarios/credor-vo.ts
/**
 * PJe-Calc v2.15.1 — CredorVO
 * Porte 1:1 de: br.jus.trt8.pjecalc.negocio.dominio.calculo.honorarios.CredorVO
 *
 * Ref Java: pjecalc-fonte/.../calculo/honorarios/CredorVO.java
 *
 * ViewObject para identificar um credor de honorário (nome + tipo/num doc).
 * `getRepresentacaoTextual()` é usado no combobox de seleção da UI;
 * `converteCredor(str)` faz o parse reverso.
 */
import { TipoDocumentoFiscalEnum } from '../../../constantes/enums';

const NAO_INFORMADO = ' não informado';

export class CredorVO {
  private nome: string | null = null;
  private tipoDocumentoFiscalCredor: TipoDocumentoFiscalEnum | null = null;
  private numeroDocumentoFiscalCredor: string | null = null;

  getNome(): string | null { return this.nome; }
  setNome(v: string | null): void { this.nome = v; }

  getTipoDocumentoFiscalCredor(): TipoDocumentoFiscalEnum | null { return this.tipoDocumentoFiscalCredor; }
  setTipoDocumentoFiscalCredor(v: TipoDocumentoFiscalEnum | null): void {
    this.tipoDocumentoFiscalCredor = v;
  }

  getNumeroDocumentoFiscalCredor(): string | null { return this.numeroDocumentoFiscalCredor; }
  setNumeroDocumentoFiscalCredor(v: string | null): void {
    this.numeroDocumentoFiscalCredor = v;
  }

  /** getRepresentacaoTextual (Java linha 39) — string p/ combobox UI. */
  getRepresentacaoTextual(): string {
    let s = '';
    s += this.nome !== null ? this.nome : `Nome${NAO_INFORMADO}`;
    s += ' - ';
    s += this.tipoDocumentoFiscalCredor !== null
      ? this.tipoDocumentoFiscalCredor
      : `Tipo de Documento${NAO_INFORMADO}`;
    s += ': ';
    s += this.numeroDocumentoFiscalCredor !== null
      ? this.numeroDocumentoFiscalCredor
      : `Número de Documento${NAO_INFORMADO}`;
    return s;
  }

  /** converteCredor (Java linha 64) — parse reverso do combobox. */
  static converteCredor(selecao: string | null): CredorVO {
    const vo = new CredorVO();
    if (!selecao || selecao === 'Preencher Dados...') return vo;
    const [nomePart, resto] = selecao.split(' - ', 2);
    if (nomePart && !nomePart.includes(NAO_INFORMADO)) {
      vo.setNome(nomePart);
    }
    if (resto) {
      const [tipoPart, numPart] = resto.split(': ', 2);
      if (tipoPart && !tipoPart.includes(NAO_INFORMADO)) {
        const enumValue = (TipoDocumentoFiscalEnum as unknown as Record<string, TipoDocumentoFiscalEnum>)[tipoPart];
        if (enumValue) vo.setTipoDocumentoFiscalCredor(enumValue);
      }
      if (numPart && !numPart.includes(NAO_INFORMADO)) {
        vo.setNumeroDocumentoFiscalCredor(numPart);
      }
    }
    return vo;
  }
}
